warehouse/warehouse/resources/dataset_spec.yaml

```diff
@@ -811,6 +811,213 @@ project:
                 grid: 180x360
                 data_types:
                   - time-series.mon
+      damip_hist-GHG:
+        start: 1850
+        end: 2014
+        ens:
+          - ens1
+          - ens2
+          - ens3
+        campaign: DECK-v1
+        science driver: Water Cycle
+        resolution:
+          1deg_atm_60-30km_ocean:
+            atmos:
+              -
+                grid: 180x360
+                data_types:
+                  - climo.mon
+                  - time-series.mon
+              -
+                grid: native
+                data_types:
+                  - model-output.mon
+                  - model-output.day
+                  - model-output.day_cosp
+                  - model-output.6hr_snap
+                  - model-output.3hr
+                  - model-output.6hr
+            land:
+              -
+                grid: native
+                data_types:
+                  - model-output.mon
+              -
+                grid: 180x360
+                data_types:
+                  - time-series.mon
+            river:
+              -
+                grid: native
+                data_types:
+                  - model-output.mon
+            ocean:
+              -
+                grid: native
+                data_types:
+                  - model-output.mon
+                  - model-output.5day_snap
+            sea-ice:
+              -
+                grid: native
+                data_types:
+                  - model-output.mon
+      damip_SSP585-GHG:
+        start: 2015
+        end: 2099
+        ens:
+          - ens1
+          - ens2
+          - ens3
+        campaign: DECK-v1
+        science driver: Water Cycle
+        resolution:
+          1deg_atm_60-30km_ocean:
+            atmos:
+              -
+                grid: 180x360
+                data_types:
+                  - climo.mon
+                  - time-series.mon
+              -
+                grid: native
+                data_types:
+                  - model-output.mon
+                  - model-output.day
+                  - model-output.day_cosp
+                  - model-output.6hr_snap
+                  - model-output.3hr
+                  - model-output.6hr
+            land:
+              -
+                grid: native
+                data_types:
+                  - model-output.mon
+              -
+                grid: 180x360
+                data_types:
+                  - time-series.mon
+            river:
+              -
+                grid: native
+                data_types:
+                  - model-output.mon
+            ocean:
+              -
+                grid: native
+                data_types:
+                  - model-output.mon
+                  - model-output.5day_snap
+            sea-ice:
+              -
+                grid: native
+                data_types:
+                  - model-output.mon
+      projection:
```

Comment:
Is the name of the experiment really "projection"?

Reply:
Good catch! I think we can use "projection_SSP585" to clarify it?

Reply:
By the way, should we use the same names for corresponding BGC SSP runs?

Reply:
`ssp585-BxRx` were used for ssp585 runs for BGC. I think here we can use `ssp585` to replace `projection`, use `ssp585-GHG` for `damip_SSP585-GHG`; and `hist-GHG` for `damip_hist-GHG`. These suggested names follows CMIP experiment naming convention. Also I understand these names in data-specs will be in the publication directory and become experiment names in search facet?

Reply:
I like it.  Yes, I try to replicate the experiment names exactly everywhere (Archive_Locator, Archive_Map, Warehouse paths, Publication Paths, and of course the dataset_spec.  Any deviations should be punishable by . . .

One area of concern is campaign (DECK, BGC, CRYO, HR).  In many cases there is no explicit mention of these, or else "model" is used near-synonymously. I always knew "projection" was a DECK-ism.  If I grep for "ssp585", I will now receive DECK + BGC material.  Not a big deal, just a change in habits.

Seriously. If we do not care about campaign in the facets, why do we say "DECK this" and "BGC that", etc?

Reply:
Like `period`, `campaign` is another custom facets that are not inferrable from data paths and data names. should `campiangn` also something be defined in dataset_specs?

Reply:
The campaign and science driver are already included, see here: https://github.com/E3SM-Project/esgfpub/blob/warehouse/warehouse/warehouse/resources/dataset_spec.yaml#L573

Perhaps it would be better to have a "custom_facets" section and list them all in one spot.

Reply:
It is not that they are not recorded in the dataset_spec - it is that they are not reflected in many other external resources, I assume because they correspond roughly with "model".  The day we have two different campaigns use the same model - there will be trouble.  Also, campaign subsumes a collection of models (CRYO -> {1.2, 1.2.1, 1.3}) so when someone says "Do X with CRYO" one is going to struggle.  When someone asks me "...something about CRYO", I will respond, "Did you mean, something about data models {1.2, 1.2.1, 1.3}?"

To my knowledge, the ONLY places you can easily list "all DECK" or "all BGC" or "all CRYO" or "all HR" datasets (without having a handy yaml-parser) is through the Archive_Map, or one of the periodic reports (AWPS, Warehouse_Status) that use the Archive_Map.

Perhaps I am wrong.  Is there a simple, one-line CLI command I can use to return "all CRYO experiment names", or "all DECK datasetIDs" or "all HR publication paths"?  If not, we should ban the mention of campaigns in all dataset and experiment discussions, as the "mapping" is distant and difficult.